pennylane/templates/subroutines/arithmetic/semi_adder.py

```diff
@@ -56,6 +61,41 @@ def _right_operator(wires, ik_is_zero=False):
     return [CNOT([ck, aux]), adjoint(TemporaryAND([ck, tk, aux]))]
 
 
+def _controlled_right_operator(
+    wires, control_wires, control_values, ik_is_zero=False, work_wires=None
+):
+    """Implement the right block in figure 4, https://arxiv.org/pdf/1709.06648"""
+
+    if not ik_is_zero:
```

Comment:
Can you add a comment here about what does "ik_is_zero" tells you? Otherwise, having args in the docstring would be better.